minecraft/1.8/src/main/java/nova/core/wrapper/mc/forge/v18/wrapper/CategoryConverter.java

```diff
@@ -0,0 +1,134 @@
+ *
+ * @author ExE Boss
+ */
+public class CategoryConverter implements NativeConverter<Category, CreativeTabs>{
+
+	public static CategoryConverter instance() {
+		return (CategoryConverter) Game.natives().getNative(Category.class, CreativeTabs.class);
+	}
+
+	@Override
+	public Class<Category> getNovaSide() {
+		return Category.class;
+	}
+
+	@Override
+	public Class<CreativeTabs> getNativeSide() {
+		return CreativeTabs.class;
+	}
+
+	@Override
+	public Category toNova(CreativeTabs creativeTab) {
+		switch (creativeTab.getTabLabel()) {
+			case "buildingBlocks":
+				return Category.BUILDING_BLOCKS;
+			case "decorations":
+				return Category.DECORATIONS;
+			case "redstone":
+				return Category.TECHNOLOGY;
+			case "transportation":
+				return Category.TRANSPORTATION;
+			case "food":
+				return Category.FOOD;
+			case "tools":
+				return Category.TOOLS;
+			case "combat":
+				return Category.COMBAT;
+			case "brewing":
+				return Category.ALCHEMY;
+			case "materials":
+				return Category.MATERIALS;
+			case "misc":
+				return Category.MISCELLANEOUS;
+			default:
+				return new Category(creativeTab.getTabLabel(), ItemConverter.instance().toNova(creativeTab.getIconItemStack()));
+		}
+	}
+
+	@Override
+	public CreativeTabs toNative(Category category) {
+		return toNative(category, new ItemStack((net.minecraft.item.Item) null));
+	}
+
+	public CreativeTabs toNative(Category category, net.minecraft.block.Block block) {
+		return toNative(category, new ItemStack(block));
+	}
+
+	public CreativeTabs toNative(Category category, net.minecraft.item.Item item) {
+		return toNative(category, new ItemStack(item));
+	}
+
+	public CreativeTabs toNative(Category category, ItemStack item) {
+		if (category.name.startsWith("nova:")) {
+			switch (category.name) {
+				case Category.BUILDING_BLOCKS_NAME:
+					return CreativeTabs.tabBlock;
+				case Category.DECORATIONS_NAME:
+					return CreativeTabs.tabDecorations;
+				case Category.TECHNOLOGY_NAME:
+					return CreativeTabs.tabRedstone;
+				case Category.TRANSPORTATION_NAME:
+					return CreativeTabs.tabTransport;
+				case Category.FOOD_NAME:
+					return CreativeTabs.tabFood;
+				case Category.TOOLS_NAME:
+					return CreativeTabs.tabTools;
+				case Category.COMBAT_NAME:
+					return CreativeTabs.tabCombat;
+				case Category.ALCHEMY_NAME:
+					return CreativeTabs.tabBrewing;
+				case Category.MATERIALS_NAME:
+					return CreativeTabs.tabMaterials;
+				case Category.MISCELLANEOUS_NAME:
+					return CreativeTabs.tabMisc;
+			}
+		}
+		Optional<CreativeTabs> first = Arrays.stream(CreativeTabs.creativeTabArray)
+			.filter(tab -> tab.getTabLabel().equals(category.name))
+			.findFirst();
+		if (first.isPresent()) {
+			return first.get();
+		} else {
+			Optional<Item> categoryItem = category.item;
+			ModCreativeTab tab = new ModCreativeTab(category.name, categoryItem.isPresent() ? Game.natives().toNative(categoryItem.get()) : item);
```

Comment:
Refactor `categoryItem.isPresent() ? Game.natives().toNative(categoryItem.get()) : item` out into a local var. This can also be written `category.item.map(Game.natives()::toNative).orElse(item)`.